src/components/UserProfile/UserProfile.jsx

```diff
@@ -290,7 +290,10 @@ const UserProfile = props => {
     }
     try {
       await props.updateUserProfile(props.match.params.userId, userProfile);
-      await props.refreshToken(userProfile._id);
+
+        if (userProfile._id == props.auth.user.userid && props.auth.user.role !== userProfile.role) {
```

Comment:
It should be === instead of ==. Thanks.

Reply:
While testing : 1. I am also getting a blank screen while accessing my profile after visiting other user profile.
2. After login, I dont see add bluesquare option even though i am the administrator.

<img width="400" alt="Screen Shot 2023-02-02 at 12 15 02 PM" src="https://user-images.githubusercontent.com/16258136/216440131-be50fd92-ad21-48ab-a52e-e3330ffdc72f.png">
<img width="400" alt="Screen Shot 2023-02-02 at 12 15 15 PM" src="https://user-images.githubusercontent.com/16258136/216440145-3dd23ffd-6af4-4c65-8ff4-50915c504e40.png">

Reply:
I am not getting any of this errors, please take a look at the video that I made, please verify to update your development branch when testing any PR.
Here is the link for the video:[Video]https://www.dropbox.com/s/naw9lqb3zu3o79f/testing_user_role_change.mp4?dl=0(url)

Reply:
I am using your branch for front end and development branch for back end. I am not getting the bluesquare + button when i login. Am i missing something here ? I have build both the applications before starting. Thanks. 

<img width="400" alt="image" src="https://user-images.githubusercontent.com/16258136/216489347-08de5898-0c22-4877-9d5b-079cd00455eb.png">

Reply:
Hi Johny, I see it is a problem in both the development branches as well. This is not related to your PR. So moving it to bugs doc. Once the === is fixed i shall approve this PR. Thanks.

Reply:
Done! take a look at it.

Reply:
Good now! Thanks.